pkg/minikube/cluster/ip.go

```diff
@@ -139,6 +139,10 @@ func HostIP(host *host.Host, clusterName string) (net.IP, error) {
 			return []byte{}, errors.Wrap(err, "Error converting VM IP address to IPv4 address")
 		}
 		return net.IPv4(vmIP[0], vmIP[1], vmIP[2], byte(1)), nil
+	case driver.VFKit:
+		vmIPString, _ := host.Driver.GetIP()
+		gatewayIPString := vmIPString[:strings.LastIndex(vmIPString, ".")+1] + "1"
+		return net.ParseIP(gatewayIPString), nil
```

Comment:
This code should really be handled by the driver directly, instead of this switch case poking at driver internals

Notice how hard it is to add extra networking options to the driver, without fiddling around in minikube code

Didn't look into what part of "vanilla" uses HostIP

But using a helper function for hyperkit sounds fair.

Reply:
I agree, i would love to see a follow up refactor if someone interested, to move this logic to the driver code not in minikube code